contrib/terraform/openstack/modules/ips/main.tf

```diff
@@ -15,7 +15,7 @@ resource "openstack_networking_floatingip_v2" "k8s_master" {
 }
 
 resource "openstack_networking_floatingip_v2" "k8s_masters" {
-  for_each   = var.number_of_k8s_masters == 0 && var.number_of_k8s_masters_no_etcd == 0 ? { for key, value in var.k8s_masters : key => value if value.floating_ip } : {}
+  for_each   = var.number_of_k8s_masters == 0 && var.number_of_k8s_masters_no_etcd == 0 ? { for key, value in var.k8s_masters : key => value if value.floating_ip && (value.reserved_floating_ip == "") } : {}
```

Comment:
Was this the one that gave you trouble when `reserved_floating_ip` is set to optional? I assume this was `value.reserved_floating_ip == null` ?

If so it's very strange that it errored out :thinking: It shouldn't be an issue.

Reply:
Yes, here is the full error if I try with the `null` check and don't set the variable for one of the `k8s_masters`:
```console
╷
│ Error: Unsupported attribute
│ 
│   on modules/ips/main.tf line 18, in resource "openstack_networking_floatingip_v2" "k8s_masters":
│   18:   for_each   = var.number_of_k8s_masters == 0 && var.number_of_k8s_masters_no_etcd == 0 ? { for key, value in var.k8s_masters : key => value if value.floating_ip && (value.reserved_floating_ip == null) } : {}
│ 
│ This object does not have an attribute named "reserved_floating_ip".
```

Reply:
Aah, right probably because it's in the for loop. 
But you should be able to use the [lookup function](https://developer.hashicorp.com/terraform/language/functions/lookup) to solve that. something like

`lookup(value, "reserved_floating_ip", "") == ""`